src/components/screens/form-edit-screen.tsx

```diff
@@ -120,55 +132,191 @@ export class FormEditScreen extends React.Component<Props, State> {
     return (
       <Grid item md={ 8 } className={ this.props.classes.formEditor }>
         <Box className={ this.props.classes.editableForm }>
-          { this.state.metaform?.title } 
+          { this.renderMainHeader() }
+          { this.state.sections?.map((section, i) => {
+            return (
+              <div key={ i } className={ this.props.classes.editableSections }>
+                { this.renderFormFields(section) }
+              </div> 
+            );
+          })}
         </Box>
       </Grid>
     )
   }
 
+  /**
+   * Renders main header
+   * 
+   * @returns Metaform main header
+   */
+  private renderMainHeader = () => (
+    <FormControl variant="outlined" className={ this.props.classes.mainHeader }>
+      <InputLabel htmlFor="mainHeaderField">{ strings.formEditScreen.formMainHeader }</InputLabel>
+      <OutlinedInput
+        label={ strings.formEditScreen.formMainHeader }
+        id="mainHeaderField"
+        color="secondary"
+        value={ this.state.metaform.title }
+        onChange={ this.handleInputTitleChange }
+      />
+    </FormControl>
+  )
+
+  /**
+   * Event handler for main header change
+   * 
+   * @param event new main header value
+   */
+  private handleInputTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newMetaform = this.state.metaform;
+    newMetaform.title = event.target.value;
+    this.setState({
+      metaform : newMetaform
+    })
+  }
+
+  /**
+   * Method for rendering form fields
+   */
+  private renderFormFields = (section : MetaformSection) => (
+    <FormControl>
+      {
+        section.fields?.map((field : MetaformField, i : number) => {
+          return (
+            <div key={ i } className={ this.props.classes.editableField }>
+              { this.renderInput(field, i) }
+            </div>
+          )
+        })
+      }
+    </FormControl>
+  );
+
+  /**
+  * Renders field's input
+  */
+  private renderInput = (field : MetaformField, i : number) => {
+  switch (field.type) {
+    case MetaformFieldType.Text:
+      return  <MetaformTextFieldComponent
+                field={ field }
+              />;
```

Comment:
same for the other ones
```suggestion
      return  (
        <MetaformTextFieldComponent
          field={ field }
        />
      );
```